index.rst

```diff
@@ -68,16 +68,16 @@ File Format
 ===========
 
 EditorConfig files are in an INI-like file format.
-In an EditorConfig file, all beginning
-whitespace on each line is considered irrelevant. Each line must be one of the
-following:
+In an EditorConfig file, all beginning whitespace on each line is considered
+irrelevant. Each line must be one of the following:
 
 - Blank: contains only whitespace characters.
 - Comment: starts with a ``;`` or a ``#``.
-   - Inserting a ``#`` or ``;`` after non-whitespace characters in a line
-     (i.e., inline) shall neither be parsed as a comment nor as part of the
-     section name, pair (defined below) key or value in which it was inserted. This may change
-     in the future; thus, is not recommended.
+   - Inserting an unescaped ``#`` or ``;`` after non-whitespace characters in
+     a line (i.e. inline) is not parsed as a comment, nor as part of
+     the section name, the key pair (see below), or the value it was inserted
+     into. This behavior may change in the future; therefore this kind of
+     insertion is not recommended.
```

Comment:
This text is still not clear to me. For the specific case where `;` is included in a property value:

```
property_name = aaaa;bbbb
```

Which of the following behaviors is expected?

1. The property value is `aaaa` (`;` and everything after it are ignored)
2. The property value is `aaaa;bbbb` (`;` has no special meaning)
3. The property value is `aaaabbbb` (`;` has no special meaning for comments, but is not parsed as part of the value)

According to literal interpretation of this text, (3) appears to be the only allowed answer; however, I do not think this is the behavior that was intended.

Reply:
I see - `;` has no special meaning, if it's not at the beginning of the line.

I guess i will rework the specification to remove negations. I think this whole paragraph is a leftover from former interpretations and implementations.

Reply:
The word "unescaped" leaves the impression there's a way for escaping. But escaping is not allowed even though some ini-parsers may provide given mechanisms. I am sorry for the confusion.

Reply:
I guess it would be easier to just say something like "what ``#`` or ``;`` after non-whitespace characters do is undefined". What do you think about the following:

> Inserting an unescaped ``#`` or ``;`` after non-whitespace characters in a line (i.e. inline) is not guaranteed to be parsed as a comment, nor as part of the section name, the key pair (see below), or the value it was inserted into. Therefore this kind of insertion leads to undefined behavior and is not recommended.

Reply:
I think the specification should not leave any way for interpretation, so a conforming core is not allowed to parse it as comment. And that's the reason why (i should come to this conclusion earlyier) i would remove that paragraph or add it as "informative" anecdote.

Reply:
I think the editorconfig and ini-parsing in general are two different things because ini-parsing is heavily context specific where ini-parsing isn't. I open a PR to o draft my idea..

Reply:
It looks this is still not clear. But we need clarification for fixing this anyoning "bug" https://github.com/dotnet/roslyn/issues/44596